File: Evaluate Reverse polish notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        vector<int> stack;
        if(tokens.size()==1)
            return stoi(tokens[0]);
        stack.push_back(stoi(tokens[0]));
        stack.push_back(stoi(tokens[1]));
        for(int i=2; i<tokens.size(); i++){
            if(tokens[i]=="+"){
                int n = stack.size()-1;
                int b= stack[n];
                int a = stack[n-1];
                stack.pop_back();
                stack.pop_back();
                stack.push_back(a+b);
            }
             else if(tokens[i]=="-"){
                int n = stack.size()-1;
                int b= stack[n];
                int a = stack[n-1];
                stack.pop_back();
                stack.pop_back();
                stack.push_back(a-b);
            }
            else if(tokens[i]=="*"){
                int n = stack.size()-1;
                int b= stack[n];
                int a = stack[n-1];
                stack.pop_back();
                stack.pop_back();
                stack.push_back(a*b);
            }
            else if(tokens[i]=="/"){
                int n = stack.size()-1;
                int b= stack[n];
                int a = stack[n-1];
                stack.pop_back();
                stack.pop_back();
                stack.push_back(a/b);
            }
            else{
               stack.push_back(stoi(tokens[i])); 
            }
        }
        return stack[0];
    }
};

File: Evaluate Reverse Polish Notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        operation = []
        operands = ["+", "-", "*", "/"]
        for token in tokens:
            if token in operands:
                opr1 = operation.pop()
                opr2 = operation.pop()
                if token == "+": 
                    operation.append(opr1 + opr2)
                elif token == "-":
                    operation.append(opr2 - opr1)
                elif token == "*":
                    operation.append(opr1 * opr2)
                else:
                    operation.append(int(opr2 / opr1))
            else:
                operation.append(int(token))
        return operation[0]

